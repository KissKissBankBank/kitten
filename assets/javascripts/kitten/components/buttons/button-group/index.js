import React from 'react'
import styled from 'styled-components'
import { Button } from '../../../components/buttons/button'
import { pxToRem } from '../../../helpers/utils/typography'
import classNames from 'classnames'

const borderRadius = pxToRem(4)
const borderSize = pxToRem(2)

const StyledButtonGroup = styled.div`
  display: flex;
  justify-content: center;

  .k-ButtonGroup__button {
    z-index: 0;
    min-width: auto;
    width: auto;

    &:not(:last-child) {
      margin-right: -${borderSize};
    }

    &:first-child {
      border-top-left-radius: ${borderRadius};
      border-bottom-left-radius: ${borderRadius};
    }

    &:last-child {
      border-top-right-radius: ${borderRadius};
      border-bottom-right-radius: ${borderRadius};
    }

    &:active,
    &:hover,
    &:focus,
    &.k-ButtonGroup__button--isActive {
      border-radius: ${borderRadius};
      z-index: 1;
    }
  }
`

export const ButtonGroup = ({ className, ...props }) => (
  <StyledButtonGroup
    role="group"
    {...props}
    className={classNames('k-ButtonGroup', className)}
  />
)

const ButtonGroupButton = ({ className, active, ...props }) => (
  <Button
    {...props}
    className={classNames('k-ButtonGroup__button', className, {
      'k-ButtonGroup__button--isActive': active,
    })}
  />
)

ButtonGroup.Button = ButtonGroupButton
