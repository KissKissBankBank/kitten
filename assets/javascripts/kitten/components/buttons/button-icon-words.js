import React from 'react'
import Button from './button'
import { HeartIcon } from 'kitten/components/icons/heart-icon'

export default class ButtonIconWords extends React.Component {
  render() {

    return (
      <Button { ...this.props } className='k-Button--icon' >
        <HeartIcon className="k-Button__icon" />
      </Button>
    )
  }
}

ButtonIconWords.defaultProps = {
  name: 'hydrogen',
  text: 'Button',
}
