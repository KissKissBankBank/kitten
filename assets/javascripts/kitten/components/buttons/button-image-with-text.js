window.ButtonImageWithText = React.createClass({
  propTypes: {
    children: React.PropTypes.string,
    onClick: React.PropTypes.func,
    isExpanded: React.PropTypes.bool,
    text: React.PropTypes.string,
    title: React.PropTypes.string,
    srcImg: React.PropTypes.string,
    widthImg: React.PropTypes.number,
    heightImg: React.PropTypes.number,
    altImg: React.PropTypes.string,
  },
  // Lifecycle
  getDefaultProps: function() {
    return {
      children: 'Toggle button',
      onClick: () => {},
    }
  },

  render: function() {
    const { className, isExpanded, text, title, classNameText,
            srcImg, widthImg, heightImg, altImg,
            ...rest } = this.props

    const buttonClassName = classNames('k-ButtonImageWithText', className)
    const textClassName = classNames('k-ButtonImageWithText__text', classNameText)

    return(
      <button className={ buttonClassName }
              aria-haspopup="true"
              aria-expanded={ isExpanded }
              title={ title }
              { ...rest }>
        <span ref="buttonImage" className="k-ButtonImage">
          <img className="k-ButtonImage__img"
               src={ srcImg }
               width={ widthImg }
               height={ heightImg }
               alt={ altImg } />
        </span>
        <p className={ textClassName }>{ text }</p>
      </button>
    )
  }
})
