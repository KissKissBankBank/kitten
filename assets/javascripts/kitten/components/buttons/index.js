export { Button } from './button'
export { ButtonIcon } from './button-icon'
export { ButtonImage } from './button-image'
export { ButtonImageWithText } from './button-image-with-text'
export { ButtonImageWithTextAndBadge } from './button-image-with-text-and-badge'
export { ButtonQuestionMarkIcon } from './button-question-mark-icon'
export { CloseButton } from './close-button'
export {
  FacebookButtonIcon,
  InstagramButtonIcon,
  LinkedinButtonIcon,
  PinterestButtonIcon,
  TwitterButtonIcon,
  YoutubeButtonIcon,
} from './social-button-icon'
export {
  FacebookButtonIconWords,
  LinkedinButtonIconWords,
  TwitterButtonIconWords,
} from './social-button-icon-words'
export { TagButton } from './tag-button'
