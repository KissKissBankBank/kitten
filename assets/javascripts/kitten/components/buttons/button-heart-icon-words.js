import React from 'react'
import { Button } from 'kitten/components/buttons/button'
import { HeartIcon } from 'kitten/components/icons/heart-icon'

export class ButtonHeartIconWords extends React.Component {
  render() {
    return (
      <Button { ...this.props } className='k-Button--icon'>
        <HeartIcon className="k-Button__icon" />
      </Button>
    )
  }
}

ButtonHeartIconWords.defaultProps = {
  name: 'hydrogen',
  text: 'Button',
}
