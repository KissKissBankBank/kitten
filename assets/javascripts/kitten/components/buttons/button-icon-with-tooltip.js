import React, { Component, Fragment } from 'react'
import Radium, { StyleRoot } from 'radium'
import PropTypes from 'prop-types'
import { ButtonIcon as ButtonIconBase } from 'kitten/components/buttons/button-icon'
import COLORS from 'kitten/constants/colors-config'

const ButtonIcon = Radium(ButtonIconBase)

export class ButtonIconWithTooltip extends Component {
  static propTypes = {
    tooltipText: PropTypes.string.isRequired,
    href: PropTypes.string,
    modifier: PropTypes.string,
    size: PropTypes.string,
  }

  static defaultProps = {
    href: '',
    modifier: '',
    size: '',
  }

  constructor(props) {
    super(props)

    this.state = {
      hover: false,
      focus: false,
    }
  }

  handleSubmitFocus = () => {
    this.setState({ focus: true })
  }

  handleSubmitBlur = () => {
    this.setState({ focus: false })
  }

  handleOnMouseEnter = () => {
    this.setState({ hover: true })
  }

  handleOnMouseLeave = () => {
    this.setState({ hover: false })
  }

  renderTooltip() {
    const { tooltipText } = this.props

    const tooltipStyle = [
      styles.tooltip.content,
      this.state.hover && styles.tooltip.content.hover,
      this.state.focus && styles.tooltip.content.focus,
    ]

    return (
      <div style={styles.tooltip}>
        <span style={tooltipStyle}>
          {tooltipText}
          <span style={styles.tooltip.content.after} />
        </span>
        {this.renderButton()}
      </div>
    )
  }

  renderButton() {
    const { children, modifier, size } = this.props

    return (
      <Fragment>
        <ButtonIcon modifier={modifier} size={size}>
          {children}
        </ButtonIcon>
      </Fragment>
    )
  }
  render() {
    const { href } = this.props

    return (
      <StyleRoot>
        <a
          href={href}
          onMouseEnter={this.handleOnMouseEnter}
          onMouseLeave={this.handleOnMouseLeave}
          onFocus={this.handleSubmitFocus}
          onBlur={this.handleSubmitBlur}
          style={{ outline: 'none' }}
        >
          {this.renderTooltip()}
        </a>
      </StyleRoot>
    )
  }
}

const backgroundColor = COLORS.primary1

const pseudoClass = {
  visibility: 'visible',
  opacity: 1,
}

const styles = {
  tooltip: {
    position: 'relative',

    content: {
      position: 'absolute',
      top: 55,
      padding: 15,
      marginLeft: -50,
      backgroundColor: backgroundColor,
      border: '2px solid #19b4fa',
      fontSize: 14,
      lineHeight: 'normal',
      fontWeight: 'regular',
      color: COLORS.background1,
      whiteSpace: 'nowrap',
      opacity: 0,
      visibility: 'hidden',
      transition: 'opacity .2s, visibility .2s',
      hover: pseudoClass,
      focus: pseudoClass,

      after: {
        position: 'absolute',
        left: '50%',
        bottom: '102%',
        height: 0,
        width: 0,
        marginLeft: -8,
        border: '10px solid transparent',
        borderBottomColor: backgroundColor,
        pointerEvents: 'none',
      },
    },
  },
}
