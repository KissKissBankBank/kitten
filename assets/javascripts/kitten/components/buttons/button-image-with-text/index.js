import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import { ButtonImage } from '../../../components/buttons/button-image'
import { Text } from '../../../components/typography/text'
import { pxToRem } from '../../../helpers/utils/typography'

const Container = styled.div`
  display: flex;
  align-items: center;
  flex: 1;
`

const TextContainer = styled.div`
  margin-left: ${pxToRem(10)};
  line-height: 1.2;

  ${({ largeGutter }) =>
    largeGutter &&
    css`
      margin-left: ${pxToRem(15)};
    `}
`

const StyledText = styled(Text)`
  display: flex;
  align-items: center;
  flex: 1;

  ${({ tag }) =>
    tag === 'p' &&
    css`
      margin: 0;
    `}
`

export const ButtonImageWithText = ({
  title,
  description,
  micro,
  tag,
  titleRegular,
  textRegular,
  largeGutter,
  avatarProps,
}) => {
  const size = micro ? 'micro' : 'tiny'
  const titleWeight = titleRegular ? 'regular' : 'light'
  const textWeight = textRegular ? 'regular' : 'light'

  return (
    <Container>
      <ButtonImage tag="span" img={avatarProps} withoutPointerEvents />

      <TextContainer largeGutter={largeGutter}>
        <StyledText tag={tag} size={size} weight={titleWeight}>
          {title}
        </StyledText>

        <StyledText tag={tag} size={size} weight={textWeight}>
          {description}
        </StyledText>
      </TextContainer>
    </Container>
  )
}

ButtonImageWithText.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  micro: PropTypes.bool,
  tag: PropTypes.string,
  titleRegular: PropTypes.bool,
  textRegular: PropTypes.bool,
  largeGutter: PropTypes.bool,
  avatarProps: PropTypes.object,
}

ButtonImageWithText.defaultProps = {
  micro: false,
  tag: 'p',
  titleRegular: false,
  textRegular: false,
  largeGutter: false,
  avatarProps: {
    src: '',
    alt: '',
  },
}
