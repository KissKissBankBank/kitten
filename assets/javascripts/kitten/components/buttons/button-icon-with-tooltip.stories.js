import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, text } from '@storybook/addon-knobs/react'
import { Container } from 'kitten/components/grid/container'
import { Grid, GridCol } from 'kitten/components/grid/grid'
import { PhoneIcon } from 'kitten/components/icons/phone-icon'
import { ButtonIconWithTooltip } from 'kitten/components/buttons/button-icon-with-tooltip'
import { Marger } from 'kitten/components/layout/marger'

storiesOf('Buttons', module)
  .addDecorator(withKnobs)
  .add('ButtonIconWithTooltip', () => {
    return (
      <Container>
        <Grid>
          <GridCol offset="1" col="3">
            <Marger top="5">
              <ButtonIconWithTooltip
                modifier={text('Modifier', 'hydrogen')}
                size={text('Size', 'tiny')}
                tooltip={text('Tooltip text', '06 00 00 00 00')}
                href={text('Tooltip link', 'tel:06 00 00 00 00')}
              >
                <PhoneIcon className="k-ButtonIcon__svg" />
              </ButtonIconWithTooltip>
            </Marger>
          </GridCol>
        </Grid>
      </Container>
    )
  })
