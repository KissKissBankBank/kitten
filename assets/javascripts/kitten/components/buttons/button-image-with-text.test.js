import React from 'react'
import renderer from 'react-test-renderer'
import { ButtonImageWithText } from 'kitten/components/buttons/button-image-with-text'

describe('<ButtonImageWithText />', () => {
  it('should match its empty snapshot', () => {
    const tree = renderer
      .create(
        <ButtonImageWithText
          title="Custom title"
          description="Custom description"
          micro={false}
          largeGutter={false}
          titleRegular={false}
          textRegular={false}
        />,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
