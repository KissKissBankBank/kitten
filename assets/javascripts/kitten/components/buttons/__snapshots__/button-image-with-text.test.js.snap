// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonImageWithText /> should match its empty snapshot 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flex": "1",
    }
  }
>
  <span
    className="k-ButtonImage k-ButtonImage--withoutPointerEvents"
    tabIndex={null}
  >
    <img
      alt=""
      className="k-ButtonImage__img"
      src="https://placehold.it/100x100/caf4fe/caf4fe"
    />
  </span>
  <div
    data-radium={true}
    style={
      Object {
        "lineHeight": "1.2",
        "marginLeft": "10px",
      }
    }
  >
    <p
      className="k-u-size-tiny k-u-weight-light"
      data-radium={true}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": "1",
          "margin": 0,
        }
      }
    >
      Custom title
    </p>
    <p
      className="k-u-size-tiny k-u-weight-light"
      data-radium={true}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": "1",
          "margin": 0,
        }
      }
    >
      Custom description
    </p>
  </div>
</div>
`;
